tests/baselines/reference/api/typescript.d.ts

```diff
@@ -3854,6 +3838,258 @@ declare namespace ts {
+     */
+    interface WatchCompilerHostOfFilesAndCompilerOptions extends WatchCompilerHost {
+        /** root files to use to generate program */
+        rootFiles: string[];
+        /** Compiler options */
+        options: CompilerOptions;
+    }
+    /**
+     * Reports config file diagnostics
+     */
+    interface ConfigFileDiagnosticsReporter {
+        /**
+         * Reports the diagnostics in reading/writing or parsing of the config file
+         */
+        onConfigFileDiagnostic: DiagnosticReporter;
+        /**
+         * Reports unrecoverable error when parsing config file
+         */
+        onUnRecoverableConfigFileDiagnostic: DiagnosticReporter;
+    }
+    /**
+     * Host to create watch with config file
+     */
+    interface WatchCompilerHostOfConfigFile extends WatchCompilerHost, ConfigFileDiagnosticsReporter {
+        /** Name of the config file to compile */
+        configFileName: string;
+        /** Options to extend */
+        optionsToExtend?: CompilerOptions;
+        /**
+         * Used to generate source file names from the config file and its include, exclude, files rules
+         * and also to cache the directory stucture
+         */
+        readDirectory(path: string, extensions?: ReadonlyArray<string>, exclude?: ReadonlyArray<string>, include?: ReadonlyArray<string>, depth?: number): string[];
+    }
+    interface Watch<T> {
+        /** Synchronize with host and get updated program */
+        getProgram(): T;
+    }
+    /**
+     * Creates the watch what generates program using the config file
+     */
+    interface WatchOfConfigFile<T> extends Watch<T> {
+    }
+    /**
+     * Creates the watch that generates program using the root files and compiler options
+     */
+    interface WatchOfFilesAndCompilerOptions<T> extends Watch<T> {
+        /** Updates the root files in the program, only if this is not config file compilation */
+        updateRootFileNames(fileNames: string[]): void;
+    }
+    /**
+     * Create the watched program for config file
+     */
+    function createWatchOfConfigFile(configFileName: string, optionsToExtend?: CompilerOptions, system?: System, reportDiagnostic?: DiagnosticReporter, reportWatchStatus?: WatchStatusReporter): WatchOfConfigFile<Program>;
+    /**
+     * Create the watched program for root files and compiler options
+     */
+    function createWatchOfFilesAndCompilerOptions(rootFiles: string[], options: CompilerOptions, system?: System, reportDiagnostic?: DiagnosticReporter, reportWatchStatus?: WatchStatusReporter): WatchOfFilesAndCompilerOptions<Program>;
+    /**
+     * Creates the watch from the host for root files and compiler options
+     */
+    function createWatchProgram(host: WatchCompilerHostOfFilesAndCompilerOptions): WatchOfFilesAndCompilerOptions<Program>;
+    /**
+     * Creates the watch from the host for config file
+     */
+    function createWatchProgram(host: WatchCompilerHostOfConfigFile): WatchOfConfigFile<Program>;
```

Comment:
can we combine just remove  `createWatchOfConfigFile` and `createWatchOfFilesAndCompilerOptions`.. we can include them in a sample in the doc, like this is how you create a watch

Reply:
Sure. I will update the PR accordingly.

Reply:
Should we be then exposing ```createWatchCompilerHost``` as part of this to have a default method that creates watch compiler host just like we have ```createCompilerHost``` exposed?

Reply:
we already have `createWatchCompilerHostOfFilesAndCompilerOptions` and `createWatchCompilerHostOfConfigFile` exposed. so it should be trivial to use these to build a new watchProgram.

Reply:
Hmm those are not exposed actually, but i will look into exposing that as overload as you mentioned in other comment.